code/object/objcollide.cpp

```diff
@@ -1023,7 +1023,7 @@ void obj_find_overlap_colliders(SCP_vector<int> &overlap_list_out, SCP_vector<in
 static SCP_vector<int> sort_list_y;
 static SCP_vector<int> sort_list_z;
 
-void obj_sort_and_collide()
+void obj_sort_and_collide(SCP_vector<int> Collision_list)
```

Comment:
This should be `&Collision_list`. Otherwise you're just copying the passed list to this one and sorting the copy. The list you intended to sort remains unchanged.

Reply:
Well that's inefficient!

Reply:
Changes made...